llms.md

```diff
@@ -0,0 +1,1039 @@
+The `spacetimedb::duration!` macro can be a convenient way to specify durations.
+
+Refer to the [official Rust Module SDK documentation on docs.rs](https://docs.rs/spacetimedb/latest/spacetimedb/attr.reducer.html#scheduled-reducers) for detailed syntax and examples, including usage of the `duration!` macro (e.g., `duration!("5s").into()` to create a `ScheduleAt`).
+
+##### Scheduled Reducer Details
+
+*   **Best-Effort Scheduling:** Scheduled reducers run on a best-effort basis and may be slightly delayed under heavy database load.
+*   **Security:** Since scheduled reducers can also be called directly by clients if not secured, it's crucial to verify the caller identity if the reducer performs sensitive operations:
+    ```rust
+    #[reducer]
+    fn scheduled_task(ctx: &ReducerContext, args: MyScheduleTable) -> Result<(), String> {
+        if ctx.sender != ctx.identity() {
+            return Err("Permission denied: Task can only be run by scheduler.".into());
+        }
+        // ... proceed with scheduled logic ...
+        Ok(())
+    }
+    ```
+
+:::info Scheduled Reducers and Connections
+Scheduled reducer calls originate from the SpacetimeDB scheduler itself, not from an external client connection. Therefore, within a scheduled reducer, `ctx.Sender` will be the module's own identity, and `ctx.ConnectionId` will not represent an external client connection.
+:::
+
+#### Row-Level Security (Client Visibility Filters)
+
+(Unstable Feature)
+
+SpacetimeDB allows defining row-level security rules using the `#[spacetimedb::client_visibility_filter]` attribute. This attribute is applied to a `const` binding of type `Filter` and defines an SQL-like query that determines which rows of a table are visible to clients making subscription requests.
+
+*   The query uses `:sender` to refer to the identity of the subscribing client.
+*   Multiple filters on the same table are combined with `OR` logic.
+*   Query errors (syntax, type errors, unknown tables) are reported during `spacetime publish`.
+
+```rust
+use spacetimedb::{client_visibility_filter, Filter, table, Identity};
+
+#[table(name = "location_state")]
+struct LocationState { #[primary_key] entity_id: u64, chunk_index: u32 }
+#[table(name = "user_state")]
+struct UserState { #[primary_key] identity: Identity, entity_id: u64 }
+
+/// Players can only see entities located in the same chunk as their own entity.
+#[client_visibility_filter]
+const PLAYERS_SEE_ENTITIES_IN_SAME_CHUNK: Filter = Filter::Sql("
+    SELECT * FROM LocationState WHERE chunk_index IN (
+        SELECT chunk_index FROM LocationState WHERE entity_id IN (
+            SELECT entity_id FROM UserState WHERE identity = :sender
+        )
+    )
+");
+```
+
+:::note Unstable Feature
+As of the time of writing, this feature might be marked as unstable or not fully implemented/enforced by the SpacetimeDB host. Check the latest official documentation for current status.
```

Comment:
I think we should spell this out specifically, with version numbers.

In SpacetimeDB 1.0, the feature was not fully implemented, and was not enforced. Modules developed for 1.0 should not use row-level security.

In SpacetimeDB 1.1, the feature is unstable. Modules developed for 1.1 may use 1.1, but must opt in to the `unstable` feature of the `spacetimedb` Cargo crate, by depending on it as `spacetimedb = { version = "1.1", features = ["unstable"] }`.